### DISCUSSION TITLE
Azure.VM.NICAttached - Support for PrivateLinkService

### DISCUSSION THREAD

**Discussion by paulmccrady** [Azure/PSRule.Rules.Azure#2560] [Ideas]
The above rule checks for NICs attached to either:

- VM
- PrivateEndpoint

Can the rule be updated to also support:

- PrivateLinkService

as we get errors with NICs attached to this service.

**Comment by BernieWhite** [comment]
@paulmccrady Yes, this should be possible. Thanks for the suggestion. Are you able to provide a sanitized sample JSON?

**Reply by paulmccrady** [reply]
@BernieWhite we have suppressed the current rule and created our own. Just add the following to the $Assert.anyOf
$Assert.HasFieldValue($TargetObject, 'Properties.privateLinkService.id')

**Reply by BernieWhite** [reply]
Added as https://github.com/Azure/PSRule.Rules.Azure/issues/2563
